File: MPP/Laborator1/src/test/java/repository/InscrieriRepositoryTest.java
package repository;

import javafx.util.Pair;
import model.Categorie;
import model.Inscriere;
import model.Proba;
import org.junit.Test;

import java.io.FileReader;
import java.io.IOException;
import java.util.List;
import java.util.Properties;
import java.util.stream.Collectors;
import java.util.stream.StreamSupport;

import static org.junit.Assert.*;

public class InscrieriRepositoryTest {
    @Test
    public void test(){
        IRepository<Pair<Integer,Integer>, Inscriere> repo;
        Inscriere inscriere=new Inscriere(1,10,"maria_avram");

        Properties prop=new Properties();

        try {
            prop.load(new FileReader("D:\\University-Courses\\MPP\\Laborator1\\src\\test\\resources\\bd.properties"));
        } catch (IOException e) {
            e.printStackTrace();
        }

        repo=new InscrieriRepository(prop);


        //size
        assertEquals(0,repo.size());

        //save
        repo.save(inscriere);
        assertEquals(1,repo.size());

        //findone
        assertEquals(inscriere,repo.findOne(new Pair<>(1,10)));

        //update
        repo.update(new Pair<>(1,10),new Inscriere(1,10,"ioana_avram"));

        List<Inscriere> lista=
                StreamSupport.stream(repo.findAll().spliterator(),false )
                        .collect(Collectors.toList());

        inscriere.setUsernameOperator("ioana_avram");
        assertEquals(inscriere,lista.get(lista.size()-1));

        repo.delete(inscriere.getID());

        //nr Probe/Participant
        repo.save(new Inscriere(1,10,"maria_avram"));
        repo.save(new Inscriere(1,11,"maria_avram"));
//        lista= StreamSupport.stream(((InscrieriRepository) repo).findProbeDupaParticipant(1).spliterator(),false)
//                        .collect(Collectors.toList());
//        assertEquals(2,lista.size());

        //max 2 probe

        try{
            repo.save(new Inscriere(1,12,"maria_avram"));
            assertEquals(2,repo.size());
        }
        catch (RepositoryException e){
            assertEquals("Participantul e deja inscris la doua probe",e.getMessage());
            assertEquals(2,repo.size());
            repo.delete(new Pair<>(1,10));
            repo.delete(new Pair<>(1,11));
        }
    }

}

File: MPP/LaboratorC#/UnitTest/InscrieriRepositoryTest.cs
using System;
using System.Collections.Generic;
using Concurs.model;
using Concurs.repository;
using Concurs.repository.utils;
using log4net.Config;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace UnitTest
{
    [TestClass]
    public class InscrieriRepositoryTest
    {
        [TestMethod]
        public void TestMethod1()
        {
            XmlConfigurator.Configure(new System.IO.FileInfo("App.config"));

            IDictionary<string, string> props = new SortedList<String, String>();

            props.Add("ConnectionString", Concurs.repository.DBUtils.GetConnectionStringByName("catalogTest"));

            InscrieriRepository repo = new InscrieriRepository(props);
            Inscriere inscriere = new Inscriere(1, 10, "maria_avram");

            //size
            Assert.AreEqual(0, repo.Size());

            //save
            repo.Save(inscriere);
            Assert.AreEqual(1, repo.Size());
            try
            {
                repo.Save(inscriere);
                Assert.Fail();
            }
            catch (System.Data.SQLite.SQLiteException e)
            {
                Assert.IsTrue(true);
            }
            catch (RepositoryException e)
            {
                Assert.AreEqual(e.Message, "Error: Nu s-a putut adauga inscrierea!");
                Assert.Fail();
            }
            //findone
            Assert.AreEqual(inscriere, repo.FindOne(new KeyValuePair<int, int>(1, 10)));

            //update
            repo.Update(new KeyValuePair<int, int>(1, 10), new Inscriere(1, 10, "ioana_avram"));

            List<Inscriere> lista = (List<Inscriere>)repo.FindAll();

            inscriere.UsernameOperator="ioana_avram";
            Assert.AreEqual(inscriere, lista[lista.Count - 1]);

            try
            {
                repo.Update(new KeyValuePair<int,int>(0,0),inscriere);
                Assert.Fail();
            }
            catch (RepositoryException e)
            {
                Assert.AreEqual(e.Message, "Error: Nu s-a putut actualiza inscrierea!");
            }

            repo.Delete(inscriere.Id);

            //nr Probe/Participant
            repo.Save(new Inscriere(1, 10, "maria_avram"));
            repo.Save(new Inscriere(1, 11, "maria_avram"));


    
            //max 2 probe

            try
            {
                repo.Save(new Inscriere(1, 12, "maria_avram"));
                Assert.AreEqual(2, repo.Size());
            }
            catch (RepositoryException e)
            {
                Assert.AreEqual("Participantul e deja inscris la doua probe", e.Message);
                Assert.AreEqual(2, repo.Size());
                repo.Delete(new KeyValuePair<int, int>(1, 10));
                repo.Delete(new KeyValuePair<int, int>(1, 11));
            }
        }
    
    }
}
